### PR TITLE
More proc macro tweaks

### PR DESCRIPTION
**Pull request by nnethercote**
A follow-up to #97004.

r? @eddyb

### CODE REVIEW

**Comment by nnethercote** [comment]
The first 12 commits are duplicates from #97004. Once it merges I will rebase and they will disappear.

File: library/proc_macro/src/bridge/server.rs [hunk 1]
```diff
@@ -7,6 +7,18 @@ macro_rules! define_handles {
         'owned: $($oty:ident,)*
         'interned: $($ity:ident,)*
     ) => {
+        static COUNTERS: HandleCounters = HandleCounters {
+            $($oty: AtomicUsize::new(1),)*
+            $($ity: AtomicUsize::new(1),)*
+        };
```
Comment (bjorn3) [inline library/proc_macro/src/bridge/server.rs:13]: cc https://github.com/rust-lang/rust/pull/97004#issuecomment-1139328275

**Comment by JohnCSimon** [comment]
ping from triage:
@nnethercote 
Returning to author to address merge conflicts. 

FYI: when a PR is ready for review, send a message containing 
`@rustbot ready` to switch to `S-waiting-on-review` so the PR is in the reviewer's backlog.

@rustbot author

**Comment by nnethercote** [comment]
This is blocked waiting on #98223 to be merged. (The author of that PR is the same as the reviewer for this PR.)

**Comment by nnethercote** [comment]
I'm closing this. The first commit is now irrelevant, I've ported the second to #121835, and the third is dubious.

**Pull request closed without merging**